File: backend/src/models/user.model.ts
import mangoose from 'mongoose';

const userSchema = new mangoose.Schema({
    name: {
        type: String,
        required: true,
    },
    email: {
        type: String,
        required: true,
        unique: true,
    },
    password: {
        type: String,
        required: true,
    },
    role: {
        type: String,
        enum: ['user', 'admin'],
        default: 'user',
    },
},{
    timestamps: true,
});

const User = mangoose.model('User', userSchema);
export default User;

File: frontend/lib/data/models/user_model.dart
class UserModel {
  final String id;
  final String name;
  final String email;
  final String token;

  UserModel({
    required this.id,
    required this.name,
    required this.email,
    required this.token,
  });

  factory UserModel.fromJson(Map<String, dynamic> json) {
    return UserModel(
      id: json["user"]["_id"],

      name: json["user"]["name"],

      email: json["user"]["email"],

      token: json["token"],
    );
  }
}
